pysat/_instrument.py

```diff
@@ -545,6 +550,53 @@ def __setitem__(self, key, new):
                     self.data[keyname] = in_data[keyname]
             # attach metadata            
             self.meta[key] = new
+
+            """
+            #print("key", key)
+            #print(new)
+                
+            if not isinstance(new, dict):
+                new = {'data': new}
+            '''
+                coordinates = {}
+                dimensions = {}
+                name = key
+                attributes = {'long_name': key, 'units':''}
+            else:
+                try:
+                    coordinates = new.pop('coords')
+                except KeyError:
+                    print("No coords given")
+                    coordinates = {}
+                try:
+                    dimensions = new.pop('dims')
+                except KeyError:
+                    print("No dims given")
+                    dimensions = {}
+                try:
+                    name = new.pop('name')
+                except KeyError:
+                    print("No name given")
+                    name = key
+                try:
+                    attributes = new.pop('attrs')
+                except KeyError:
+                    print("No attrs given")
+                    attributes = {'long_name': key, 'units':''}
+            '''
+                # TODO: Support for encoding and Fastpath ???
+            in_data = new.pop('data')
+            # TODO: Set a common coordinates for the DataSet ? (Not individual DataArrays)
+            # This can be time or mtime, lat, lon, ilev or Z, LBC (needed?)
+            '''
+            # TODO: Add support for methods like assign_attrs(*args, **kwargs), assign_coords(**kwargs), etc.?
+            self.data[key] = xr.DataArray(data = in_data, coords = coordinates,
+                     dims = dimensions, name = name, attrs = attributes)
+            '''
+            self.data[key] = xr.DataArray(data = in_data)
```

Comment:
This is not wrong, but the style in the code is not to have white space for kwarg assignment [e.g., xr.DataArray(data=in_data)]

Reply:
It could very well be great to be able to have any combination of coords, dimensions, etc within the dataset. However, if we can't make any assumptions about the data we have then it becomes very difficult to do anything at all. I'd like to hold off on changes to these assumptions until I have a chance to work with xarray data in a practical context, the TIEGCM data.